docs/study-def-tricks.md

```diff
@@ -319,4 +319,63 @@ study = StudyDefinition(
     ),
     **common_variables
 )
-```
\ No newline at end of file
+```
+
+## Extracting ethnicity across multiple time points
+
+Querying ethnicity can take a long time as a result of having to map from individual codes to broad categories. This is ok if you only extract ethnicity at one time point, but if you are running your study definition across multiple time points it can slow down the cohort extraction. As ethnicity is generally expected to remain constant over time, you can instead extract ethnicity once (taking the latest recorded ethnicity code for each patient) and then map these to the files generated by your main study definition for each time point.
+
+An example study definition for extracting ethnicity is shown below. Note that the study population should contain **all** patients.
+
+`study_definition_ethnicity.py`
+```py
+    study = StudyDefinition(
+        default_expectations={
+            "date": {"earliest": "1900-01-01", "latest": "today"},
+            "rate": "uniform",
+        },
+
+        population=patients.all(),
+
+        # ETHNICITY IN 6 CATEGORIES
+        eth=patients.with_these_clinical_events(
+            ethnicity_codes,
```

Comment:
add comment here saying where these `ethnicity_codes` come from or what's in this codelist? or can we expect that this is known if people look at this section?

Reply:
I think this probably deserves its own section somewhere, defining ethnicity is a bit of a can of worms in itself. What to use for EMIS, adding in hospital recorded ethnicity etc. I'll make a note to put it somewhere else